core/domain/classifier_services.py

```diff
@@ -124,6 +124,35 @@ def classify_string_classifier_rule(state, normalized_answer):
 
     return None
 
+def check_classification_condition(state):
```

Comment:
I see. I think you probably do want to go with adding a method to the 'state' domain object here, then -- something like `if state.can_undergo_classification()`.